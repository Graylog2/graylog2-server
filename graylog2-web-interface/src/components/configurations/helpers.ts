const getConfig = (configType, configuration) => configuration?.[configType] ?? null;

export { getConfig };
export default getConfig;
