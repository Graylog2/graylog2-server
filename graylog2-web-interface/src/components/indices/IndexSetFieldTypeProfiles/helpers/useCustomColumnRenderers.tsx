/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React, { useMemo } from 'react';

import type { CustomFieldMapping, IndexSetFieldTypeProfile } from 'components/indices/IndexSetFieldTypeProfiles/types';
import CustomFieldMappingsCell from 'components/indices/IndexSetFieldTypeProfiles/cells/CustomFieldMappingsCell';
import IndexSetsCell from 'components/indices/IndexSetFieldTypeProfiles/cells/IndexSetsCell';

const useCustomColumnRenderers = (normalizedIndexSetsTitles: Record<string, string>) => useMemo(() => ({
  attributes: {
    custom_field_mappings: {
      renderCell: (customFieldTypes: Array<CustomFieldMapping>, profile: IndexSetFieldTypeProfile) => (
        <CustomFieldMappingsCell profile={profile}
                                 customFieldTypes={customFieldTypes} />
      ),
      staticWidth: 200,
    },
    index_set_ids: {
      renderCell: (indexSetIds: Array<string>) => <IndexSetsCell indexSetIds={indexSetIds} normalizedIndexSetsTitles={normalizedIndexSetsTitles} />,
    },
  },
}), [normalizedIndexSetsTitles]);

export default useCustomColumnRenderers;
