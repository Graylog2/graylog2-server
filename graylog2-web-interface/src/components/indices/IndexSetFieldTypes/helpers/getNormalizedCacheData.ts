/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */

import type { Query } from '@tanstack/react-query';
import keyBy from 'lodash/keyBy';
import reduce from 'lodash/reduce';
import extend from 'lodash/extend';

import type { IndexSetFieldTypesQueryData, IndexSetFieldType } from 'components/indices/IndexSetFieldTypes/types';

const getNormalizedCacheData = (queryCache: Array<Query<IndexSetFieldTypesQueryData>>): Record<string, IndexSetFieldType> => {
  const cacheData: Array<Record<string, IndexSetFieldType>> = queryCache.map(({ state }) => keyBy(state?.data?.list, 'id'));
  const mergedCache: Record<string, IndexSetFieldType> = reduce<Record<string, IndexSetFieldType>>(cacheData, extend);

  return mergedCache;
};

export default getNormalizedCacheData;
