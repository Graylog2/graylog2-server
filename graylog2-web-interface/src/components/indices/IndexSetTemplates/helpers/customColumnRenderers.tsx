/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import React from 'react';

import TitleCell from 'components/indices/IndexSetTemplates/cells/TitleCell';
import BuiltInCell from 'components/indices/IndexSetTemplates/cells/BuiltInCell';
import type { IndexSetTemplate } from 'components/indices/IndexSetTemplates/types';
import type { ColumnRenderers } from 'components/common/EntityDataTable';

const customColumnRenderers: ColumnRenderers<IndexSetTemplate> = {
  attributes: {
    title: {
      renderCell: (title: string, template: IndexSetTemplate) => (
        <TitleCell title={title}
                   id={template.id}
                   isDefault={template.default}
                   isEnabled={template.enabled} />
      ),
    },
    built_in: {
      renderCell: (built_in: boolean, _template) => (
        <BuiltInCell builtIn={built_in} />
      ),
    },
  },
};

export default customColumnRenderers;
