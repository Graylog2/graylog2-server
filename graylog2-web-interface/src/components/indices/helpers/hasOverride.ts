import type { IndexSetFieldType } from 'components/indices/IndexSetFieldTypes/hooks/useIndexSetFieldType';

const hasOverride = (fieldType: IndexSetFieldType) => fieldType.origin === 'OVERRIDDEN_PROFILE' || fieldType.origin === 'OVERRIDDEN_INDEX';

export default hasOverride;
