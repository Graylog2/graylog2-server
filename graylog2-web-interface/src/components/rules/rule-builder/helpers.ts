/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
import type { BlockDict, RuleBuilderRule } from './types';

const getDictForFunction = (dict: BlockDict[], functionName: string) : BlockDict | undefined => (
  dict?.find((entry) => entry.name === functionName)
);

const jsonifyText = (text: string): string => {
  try {
    JSON.parse(text);

    return text;
  } catch {
    try {
      const rawMessageToJson = `{"${
        text
          .trim()
          .replace(/^\s*\n/gm, '')
          .replace(/"/g, '\\"')
          .split('\n')
          .map((line) => line.trim().split(':').map((keyValue) => keyValue.trim()))
          .filter((keyValue) => keyValue[0] && keyValue[1])
          .map((keyValue) => `${keyValue[0].replace(/\\"/g, '').trim()}":"${keyValue.slice(1).join(':').trim()}`)
          .join('","')
      }"}`;

      JSON.parse(rawMessageToJson);

      return rawMessageToJson;
    } catch {
      return JSON.stringify({ message: text });
    }
  }
};

const hasRuleBuilderErrors = (rule: RuleBuilderRule): boolean => {
  if (rule?.rule_builder?.errors?.length > 0) return true;

  if (rule?.rule_builder?.actions?.some(((action) => action.errors?.length > 0))) {
    return true;
  }

  if (rule?.rule_builder?.conditions?.some(((condition) => condition.errors?.length > 0))) {
    return true;
  }

  return false;
};

export { getDictForFunction, jsonifyText, hasRuleBuilderErrors };
