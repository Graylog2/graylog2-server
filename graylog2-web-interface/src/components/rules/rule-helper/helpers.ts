/*
 * Copyright (C) 2020 Graylog, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the Server Side Public License, version 1,
 * as published by MongoDB, Inc.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * Server Side Public License for more details.
 *
 * You should have received a copy of the Server Side Public License
 * along with this program. If not, see
 * <http://www.mongodb.com/licensing/server-side-public-license>.
 */
const niceType = (typeName) => typeName.replace(/^.*\.(.*?)$/, '$1');

const functionSignature = (descriptor) => {
  const args = descriptor.params.map((p) => (p.optional ? `[${p.name}]` : p.name));

  return `${descriptor.name}(${args.join(', ')}) : ${niceType(descriptor.return_type)}`;
};

export { functionSignature, niceType };
