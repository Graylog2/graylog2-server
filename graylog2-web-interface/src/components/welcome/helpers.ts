export const typeLinkMap = {
  dashboard: { link: 'DASHBOARDS_VIEWID' },
  search: { link: 'SEARCH_VIEWID' },
};

export const DEFAULT_PAGINATION = {
  per_page: 5,
  page: 1,
  count: 0,
  total: 0,
};
