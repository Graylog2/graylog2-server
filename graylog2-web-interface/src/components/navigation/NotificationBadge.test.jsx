import React from 'react';
import { mount } from 'enzyme';
import { CombinedProviderMock, StoreMock } from 'helpers/mocking';

jest.mock('stores/connect', () => x => x);

describe('NotificationBadge', () => {
  let notifications;
  let NotificationBadge;

  beforeEach(() => {
    notifications = {
      NotificationsActions: { list: jest.fn() },
      NotificationsStore: StoreMock('listen'),
    };
    const combinedProviderMock = new CombinedProviderMock({
      Notifications: notifications,
    });

    jest.doMock('injection/CombinedProvider', () => combinedProviderMock);

    // eslint-disable-next-line global-require
    NotificationBadge = require('./NotificationBadge');
  });

  it('triggers update of notifications', () => {
    const { NotificationsActions } = notifications;

    mount(<NotificationBadge />);

    expect(NotificationsActions.list).toHaveBeenCalled();
  });

  it('renders nothing when there are no notifications', () => {
    const wrapper = mount(<NotificationBadge total={0} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders count when there are notifications', () => {
    const wrapper = mount(<NotificationBadge total={42} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('updates notification count when triggered by store', () => {
    const wrapper = mount(<NotificationBadge total={42} />);

    expect(wrapper.find('span#notification-badge')).toHaveText('42');

    wrapper.setProps({ total: 23 });

    expect(wrapper.find('span#notification-badge')).toHaveText('23');
  });

  it('does not rerender when number of notifications does not change', () => {
    NotificationBadge.prototype.render = jest.fn(NotificationBadge.prototype.render);
    const wrapper = mount(<NotificationBadge total={42} />);

    expect(wrapper.find('span#notification-badge')).toHaveText('42');

    wrapper.setProps({ total: 42 });

    expect(wrapper.find('span#notification-badge')).toHaveText('42');
    expect(wrapper.find('NotificationBadge').get(0).type.prototype.render).toHaveBeenCalledTimes(1);
  });
});
