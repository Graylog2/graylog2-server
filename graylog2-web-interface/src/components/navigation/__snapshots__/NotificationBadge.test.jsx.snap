// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBadge renders count when there are notifications 1`] = `
<NotificationBadge
  total={42}
>
  <Nav
    bsClass="nav"
    justified={false}
    navbar={true}
    pullLeft={false}
    pullRight={false}
    stacked={false}
  >
    <ul
      className="nav navbar-nav"
      role={null}
    >
      <LinkContainer
        action="push"
        key=".0"
        onSelect={null}
        onlyActiveOnIndex={false}
        to="/system/overview"
      >
        <InactiveNavItem
          action="push"
          className="notification-badge-link"
          onClick={[Function]}
          onSelect={null}
        >
          <NavItem
            action="push"
            active={false}
            className="notification-badge-link"
            disabled={false}
            onClick={[Function]}
            onSelect={null}
          >
            <li
              className="notification-badge-link"
              role="presentation"
            >
              <SafeAnchor
                action="push"
                componentClass="a"
                disabled={false}
                onClick={[Function]}
              >
                <a
                  action="push"
                  href="#"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="button"
                >
                  <Badge
                    bsClass="badge"
                    className="badgeDanger"
                    id="notification-badge"
                    pullRight={false}
                  >
                    <span
                      className="badgeDanger badge"
                      id="notification-badge"
                    >
                      42
                    </span>
                  </Badge>
                </a>
              </SafeAnchor>
            </li>
          </NavItem>
        </InactiveNavItem>
      </LinkContainer>
    </ul>
  </Nav>
</NotificationBadge>
`;

exports[`NotificationBadge renders nothing when there are no notifications 1`] = `
<NotificationBadge
  total={0}
/>
`;
